File: python-sensitiveinfodisclosure/vulnerable12.py
import os
import openai

openai.api_key = os.environ["OPENAI_API_KEY"]


def debug_payment_issue(order_id):
    payment_token = os.environ["PAYMENT_API_TOKEN"]
    response = openai.ChatCompletion.create(
        model="gpt-4",
        messages=[
            {"role": "user", "content": f"Debug payment for order {order_id}. API token used: {payment_token}"}
        ],
    )
    return response.choices[0].message.content


if __name__ == "__main__":
    print(debug_payment_issue("ord_123"))

File: javascript-unboundedconsumption/vulnerable12.js
const OpenAI = require("openai");

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

async function generateResponse(userText) {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [{ role: "user", content: userText }],
        max_tokens: null,
    });
    return completion.choices[0].message.content;
}

module.exports = { generateResponse };
